### ISSUE TITLE
[Documents] Bricks: When using complex bricks with blocks inside, bricks get lost on saving

### ISSUE THREAD

**Issue by HaaseIT** [pimcore/pimcore#9421]
# Bug Report

### Expected behavior

Bricks don't get lost

### Actual behavior

Bricks get lost on document save

### Steps to reproduce

I would very much prefer to give you access to a staging host where you can reproduce the issue including the source-code of the project.

### Code example
```twig
<div
    class="container brickVertriebspartner brickMarginTop-{{ pimcore_select('brickMarginTop').getData() }} brickMarginBottom-{{ pimcore_select('brickMarginBottom').getData() }}">
    <div class="row">
        <div class="col-12">
            <h3>
                {% if editmode %}
                    {{ pimcore_input('headline', {'placeholder': 'Überschrift'}) }}
                {% else %}
                    {{ pimcore_input('headline').getData()|raw }}
                {% endif %}
            </h3>
        </div>
    </div>
    {% set block = pimcore_block('infoboxesBlock', {'manual' : true}).start() %}
    <li class="row">
        <ul class="haendlerlist">
            {% for i in pimcore_iterate_block(block) %}
                <li class="haendlerlistitem col-12 col-sm-6 col-md-6 col-lg-4">
                    {% do block.blockConstruct() %}{% do block.blockStart(false) %}
                    {% do block.blockControls() %}
                    <div class="haendler">
                        <div class="country">{{ pimcore_input('country') }}</div>
                        {{ pimcore_image('logo') }}
                        <div class="name">
                            {% if editmode %}
                                {{ pimcore_input('name', {'placeholder': 'Name'}) }}
                            {% else %}
                                {{ pimcore_input('name').getData()|raw }}
                            {% endif %}
                        </div>
                        <div class="address">
                            {{ pimcore_wysiwyg('address') }}

                            {% if editmode %}
                                {{ pimcore_link('phone', {'class': 'phone'}) }}<br>
                                {{ pimcore_link('web', {'target': '_blank'}) }}<br>
                                {{ pimcore_link('email') }}
                            {% else %}
                                {% if not pimcore_link('phone').isEmpty() %}
                                    {{ pimcore_link('phone', {'class': 'phone'}) }}
                                {% endif %}
                                {% if not pimcore_link('web').isEmpty() %}
                                    {{ pimcore_link('web', {'target': '_blank'}) }}
                                {% endif %}
                                {% if not pimcore_link('email').isEmpty() %}
                                    {{ pimcore_link('email') }}
                                {% endif %}
                            {% endif %}
                        </div>
                        {% for i in pimcore_block('products').iterator %}
                            <div class="products">
                                {% if editmode %}
                                    {{ pimcore_input('product', {'placeholder': 'Produkt'}) }}
                                {% else %}
                                    {{ pimcore_input('product').getData()|raw }}
                                {% endif %}
                            </div>
                        {% endfor %}
                    </div>
                    {% do block.blockEnd() %}{% do block.blockDestruct() %}
                </li>
            {% endfor %}
        </ul>
</div>
{% do block.end() %}
</div>
```
I have 3 of these bricks in a document. The first brick has 9 Blocks, the second 14, the third has 4.

The first brick always survives the save, the second and third get lost.

I was able to save the page with all bricks due to extJS session, by re-adding the bricks, but now I added two more bricks on the top of the document (headline and some text), now it always breaks on save.

This was last tested on 10.0.4

Feel free to contact me at mail@haase-it.com

Edit:
It appears, that there is a general issue, when a brick with a block in it is present more than once on a page.
I just had the case, that in the second instance of this fairly simple brick, the images were lost on save. After adding the images once again, it saved fine.

Source:
```twig
<div class="container brickMitarbeiter brickMarginTop-{{ pimcore_select('brickMarginTop').getData() }} brickMarginBottom-{{ pimcore_select('brickMarginBottom').getData() }}">
    <div class="row">
        {% for i in pimcore_block('mitarbeiter').iterator %}
            <div class="col-12 col-sm-6 col-md-6 col-lg-4">
                <div class="mitarbeiter">
                    {{ pimcore_image('mitarbeiterBild') }}
                    <div class="name">{{ pimcore_input('name', {'placeholder': 'Name'}) }}</div>
                    <div class="funktion">{{ pimcore_input('funktion', {'placeholder': 'Funktion'}) }}</div>
                    <div class="telefon">{{ pimcore_link('telefon') }}</div>
                    <div class="email">{{ pimcore_link('email') }}</div>
                </div>
            </div>
        {% endfor %}
    </div>
</div>
```

**Comment by fashxp** [comment]
Could you try to create and share a very simple example based on our demo so we can reproduce the issue?

Or even better - try to debug and provide a fix 😉

Thank you ...

**Comment by HaaseIT** [comment]
I am not an ExtJS expert, I am more of a backend-developer.

I can have a look but since I don't know much about the handling here, this might take a couple of weeks.

**Comment by fashxp** [comment]
simple example based on the demo would also help

**Comment by HaaseIT** [comment]
I think, this might be one of the cases that cannot be reliably reproduced with a simple example.

Same here, will take me a few weeks to make time.

**Comment by thgross** [comment]
Here is a very simple example for this bug:

```twig
{% set block = pimcore_block('myblock') %}

{% for i in block.iterator %}
    <h2>{{ pimcore_input('myblockHeader') }}</h2>
{% endfor %}
```
**This bug only appears when, like in this example, an editable ('myblockHeader') starts with the same name as the block ('myblock'),**

if you change `pimcore_block('myblock')` to `pimcore_block('yourblock')` **OR** `pimcore_input('myblockHeader')` to `pimcore_input('yourblockHeader')`, everything works correctly.

Like in @HaaseIT s example: `mitarbeiterBild` inside a block names `mitarbeiter` triggers the bug.

**Comment by dvesh3** [comment]
Regression of https://github.com/pimcore/pimcore/pull/9225

**Comment by brusch** [comment]
Fixed by #9490
